[vinc3leong] iP

vinc3leong opened the pull request:
**Skibidi: Task Manager Chatbot**

_"Yooo, Skibidi gonna keep you on point, no cap!" – Gen Alpha (probably)_

Skibidi, the ultimate task manager chatbot, frees your mind from remembering the grind. Whether it's homework, projects, or just life stuff, Skibidi speaks your language—literally. It's:

- 📝 **Text-based** – Just type, no need for fancy menus.
- 🎓 **Easy to learn** – Skibidi speaks simple.
- ⚡ **FAST SUPER FAST** – Don’t blink, Skibidi’s got it handled.

 **How to Get Skibidi:**

1. Download Skibidi from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. Double-click to open (easy peasy).
3. Add your tasks (Skibidi gets it done).
4. Let Skibidi manage your life, fam. 😉

And did we mention? It's **FREE**.

---

## Features:

- ✅ Managing tasks
- ⏳ Managing deadlines (coming soon)
- ⏰ Reminders (coming soon)

---

Give Skibidi a try and keep it sigma💯with your to-do list!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

vinc3leong committed:
changed chatbot name

vinc3leong committed:
skibidi echoes back input

vinc3leong committed:
skibidi listing function

vinc3leong committed:
mark and unmark tasks

vinc3leong committed:
implemented different types of tasks

vinc3leong committed:
text ui testing

vinc3leong committed:
error handling

vinc3leong committed:
delete tasks

vinc3leong committed:
Save tasks to hard drive and merge branch with master

vinc3leong committed:
record tasks with date and time

vinc3leong committed:
extract related code as classes

vinc3leong committed:
Merge branch 'branch-A-MoreOOP'

vinc3leong committed:
Divide classes into packages

vinc3leong committed:
Merge branch 'branch-A-Packages'

vinc3leong committed:
Merge remote-tracking branch 'origin/add-gradle-support'

vinc3leong committed:
Add JUnit tests

vinc3leong committed:
Merge branch 'branch-A-JUnit'

vinc3leong committed:
Package Skibidi as Jar file

vinc3leong committed:
Add JavaDoc Comments

vinc3leong committed:
Merge branch 'branch-A-JavaDoc'

vinc3leong committed:
Follow the coding standard

vinc3leong committed:
Merge branch 'branch-A-CodingStandard'

vinc3leong committed:
Find tasks using keywords

vinc3leong committed:
Merge branch 'branch-A-level-9'

yongqqqq commented on src/main/java/skibidi/DeadlineCommand.java:
```diff
@@ -0,0 +1,51 @@
+    private final String by;
+    private final MilitaryTime time;
+
+    public DeadlineCommand(String[] input) throws SkibidiException, DateTimeParseException {
```
Please add the header comment for this public method. I noticed the same issue for other public methods as well.

yongqqqq commented on src/main/java/skibidi/Parser.java:
```diff
@@ -0,0 +1,93 @@
+    public static Command parse(String input) throws SkibidiException {
+        String[] words = input.split(" ");
+        String command = words[0];
+        switch (command) {
+            case "bye":
+                return new ExitCommand();
+            case "list":
+                return new ListCommand();
+            case "delete":
+                return new DeleteCommand(words);
+            case "todo":
+                return new TodoCommand(words);
+            case "deadline":
+                return new DeadlineCommand(words);
+            case "event":
+                return new EventCommand(words);
+            case "mark":
+                return new MarkCommand(words);
+            case "unmark":
+                return new UnmarkCommand(words);
+            case "find":
+                return new FindCommand(words);
+            default:
+                return new InvalidCommand();
+        }
```
Please edit this switch statement to follow the coding standard

yongqqqq commented on src/main/java/skibidi/Parser.java:
```diff
@@ -0,0 +1,93 @@
+            case "T":
+                boolean isDone = words[1].equals("1");
+                String description = words[2];
+                Todo todo = new Todo(description);
+                if (isDone) {
+                    todo.markAsDone();
+                    return todo;
+                } else {
+                    return todo;
+                }
+            case "D":
+                isDone = words[1].equals("1");
+                description = words[2];
+                String by = words[3];
+                MilitaryTime time = new MilitaryTime(TimeUtil.convertStandardToMilitary(words[4]));
+                Deadline deadline = new Deadline(description, by, time);
+                if (isDone) {
+                    deadline.markAsDone();
+                    return deadline;
+                } else {
+                    return deadline;
+                }
+            case "E":
+                isDone = words[1].equals("1");
+                description = words[2];
+                String from = words[3];
+                MilitaryTime startTime = new MilitaryTime(TimeUtil.convertStandardToMilitary(words[4]));
+                String to = words[5];
+                MilitaryTime endTime = new MilitaryTime(TimeUtil.convertStandardToMilitary(words[4]));
+                Event event = new Event(description, from, startTime, to, endTime);
+                if (isDone) {
+                    event.markAsDone();
+                    return event;
+                } else {
+                    return event;
+                }
+            default:
+                throw new SkibidiException("Error: Invalid task type detected in file.");
+        }
+    }
```
Please edit this switch statement to follow the coding standard

yongqqqq commented on src/main/java/skibidi/EventCommand.java:
```diff
@@ -0,0 +1,54 @@
+    private final MilitaryTime startTime;
+    private final MilitaryTime endTime;
+
+    public EventCommand(String[] input) throws SkibidiException {
+        super(COMMAND_WORD);
+        if (input.length < 8) {
+            throw new SkibidiException("Error: event description must not be empty. " +
+                    "\nTo add an event task, use the format: event <description> /from <date> <time> /to <date> <time>");
+        }
+
+        if (Arrays.stream(input).noneMatch(s -> s.equals("/from")) ||
+                Arrays.stream(input).noneMatch(s -> s.equals("/to"))) {
+            throw new SkibidiException("Error: event description must not be empty. " +
+                    "\nTo add an event task, use the format: event <description> /from <date> <time> /to <date> <time>");
+        }
+
+        int startFromIndex = Arrays.asList(input).indexOf("/from") + 1;
+        int endsAtIndex = Arrays.asList(input).indexOf("/to") + 1;
+        LocalDate startFrom = LocalDate.parse(input[startFromIndex]);
+        String formattedStartFrom = startFrom.format(DateTimeFormatter.ofPattern("MM dd yyyy"));
+        LocalDate endsAt = LocalDate.parse(input[endsAtIndex]);
+        String formattedEndsAt = endsAt.format(DateTimeFormatter.ofPattern("MM dd yyyy"));
+
+        this.description = input[1];
+        this.from = formattedStartFrom;
+        this.to = formattedEndsAt;
+        this.startTime = new MilitaryTime(input[startFromIndex + 1]);
+        this.endTime = new MilitaryTime(input[endsAtIndex + 1]);
+
+    }
+
```
Please add the header comment for this public method. I noticed the same issue for other public methods as well.

yongqqqq commented on src/main/java/skibidi/TimeUtil.java:
```diff
@@ -0,0 +1,24 @@
+/**
+ * Represents a utility class for converting time between standard and military time.
+ */
+public class TimeUtil {
```
The separation of time conversion logic from other classes here is good.

yongqqqq reviewed (commented):
Overall good job! Just a few nits to fix.

1.  Several public methods have missing JavaDoc comments.
2.  Suggest further packaging the classes for better visibility.

I have only highlighted some of the methods, it would be good to look through the codebase again.

vinc3leong committed:
no message

vinc3leong committed:
organise skibidi into more packages and add a stylecheck

vinc3leong committed:
Merge branch 'branch-A-Checkstyle'

vinc3leong committed:
Add a GUI to Skibidi

vinc3leong committed:
Merge branch 'branch-level-10'

vinc3leong committed:
Add assertions to MainWindow and DeadlineCommand

- Added assertions in MainWindow to ensure user input and response
are not null or empty before processing.

- Added assertions in DeadlineCommand to validate
the input array is not null and has the required number of elements.

- These assertions help in identifying potential issues early
during development by catching invalid states and inputs.

vinc3leong committed:
Refactor and improve code quality

- Abstracted parts of the code into separate methods for better
readability and maintainability.

- Removed unnecessary method calls and streamlined
logic in DeadlineCommand.

- Ensured consistent formatting and adherence
to coding standards across the project.

vinc3leong committed:
Merge pull request #1 from vinc3leong/branch-A-Assertion

Add assertions to MainWindow and DeadlineCommand

vinc3leong committed:
Merge branch 'master' of https://github.com/vinc3leong/ip

vinc3leong committed:
Merge branch 'master' into branch-A-CodeQuality

vinc3leong committed:
Merge pull request #2 from vinc3leong/branch-A-CodeQuality

Refactor and improve code quality

vinc3leong committed:
Setup CI

vinc3leong committed:
Add archive feature

vinc3leong committed:
Merge branch 'branch-archive'

vinc3leong committed:
Include a Ui.png

vinc3leong committed:
Update README

vinc3leong committed:
Improve Skibidi's response

- improve archive() load() and loadArchive() methods in Storage class to
create file at the specified directory if file doesn't exist

- change list command to generate a different response when there are no
available task in the task list

vinc3leong committed:
update Skibidi download link in README.md

vinc3leong committed:
Update User Guide in README.md

vinc3leong committed:
Fix formatting of user guide in README.md
